File: frontend/src/features/actividades/realizar_devolucion_ep/models/realizar_devolucion_ep.ts
import type { Auditoria } from '@/models/Auditoria';

export interface RealizarDevolucionEP extends Auditoria {
  id: number;
  id_expediente: number;
  id_actividad: string;
  requiere_escalamiento_comercial: string | null;
  tipologia: string | null;
  casuistica: string | null;
  accion_a_seguir: string | null;
  observaciones: string | null;
}

export interface DictamenPrevio {
  area: string;
  tipologia: string | null;
  casuistica: string | null;
  observaciones: string | null;
}

export interface GetByExpedienteResponse {
  formulario: RealizarDevolucionEP;
  dictamenes_previos: DictamenPrevio[];
}

export const EMPTY_REALIZAR_DEVOLUCION_EP = (id_expediente: number): RealizarDevolucionEP => ({
  id: 0, id_expediente, id_actividad: 'BBVA_ESCRITURACION_REALIZAR_DEVOLUCION_EP',
  requiere_escalamiento_comercial: 'NO', tipologia: null, casuistica: null, accion_a_seguir: null, observaciones: null,
  is_active: true, row_status: true, created_by: 0, created_date: '', modified_by: null, modified_date: null,
});

export const ACCIONES_A_SEGUIR = [
  { code: 'FIRMAR_ESCRITURA', description: 'Firmar Escritura Cliente' },
  { code: 'FIRMAR_REP_LEGAL', description: 'Firmar Rep. Legal' },
  { code: 'EP_REGISTRADAS', description: 'Realizar EP Registradas' },
];

File: backend/Multibanca.Domain.Models/Multibanca/BBVA/Escrituracion/realizar_devolucion_ep.cs
namespace Multibanca.Domain.Models.Multibanca.BBVA.Escrituracion;

public class realizar_devolucion_ep
{
    public long     id { get; set; }
    public long     id_expediente { get; set; }
    public string?  id_actividad { get; set; }
    public string?  requiere_escalamiento_comercial { get; set; }
    public string?  tipologia { get; set; }
    public string?  casuistica { get; set; }
    public string?  accion_a_seguir { get; set; }
    public string?  observaciones { get; set; }

    public bool      is_active { get; set; }
    public bool      row_status { get; set; }
    public int       created_by { get; set; }
    public DateTime  created_date { get; set; }
    public int?      modified_by { get; set; }
    public DateTime? modified_date { get; set; }
}
